ext/dom/xml_document.c

```diff
@@ -250,42 +250,38 @@ PHP_METHOD(Dom_XMLDocument, createFromFile)
 	load_from_helper(INTERNAL_FUNCTION_PARAM_PASSTHRU, DOM_LOAD_FILE);
 }
 
+static int php_new_dom_write_smart_str(void *context, const char *buffer, int len)
+{
+	smart_str *str = context;
+	smart_str_appendl(str, buffer, len);
+	return len;
+}
```

Comment:
Use `size_t` instead of `int`?

Reply:
The interface is defined by libxml, so it has to be int.